### PR TITLE
New eslint rules 4

### PR DESCRIPTION
**Pull request by SelyankinFyodor**
fix

spaced-comment
space-unary-ops
prefer-destructuring
object-curly-newline
object-curly-spacing
operator-linebreak
spaced-comment
space-unary-ops
padded-blocks
no-useless-escape
no-multi-spaces
no-useless-return
no-extra-boolean-cast
no-undef
max-len
no-plusplus
no-useless-concat
no-return-assign
no-restricted-syntax

### CODE REVIEW

File: src/core/__tests__/template.test.tsx [hunk 1]
```diff
@@ -15,17 +14,19 @@ class ComponentWithTemplates extends TestComponent {
   }];
 }
 
-function renderTemplate(name: string, model?: any, container?: any, index?: number, onRendered?: () => void): Element {
+function renderTemplate(
+  name: string, model?: any, container?: any, index?: number, onRendered?: () => void): Element {
```
Comment (LazyLahtak) [inline src/core/__tests__/template.test.tsx:18]:
```suggestion
  name: string,
  model?: any,
  container?: any,
  index?: number,
  onRendered?: () => void,
): Element {
```

File: src/core/__tests__/template.test.tsx [hunk 2]
```diff
@@ -15,17 +14,19 @@ class ComponentWithTemplates extends TestComponent {
...
+  const { render } = WidgetClass.mock.calls[0][1].integrationOptions.templates[name];
 
   return render({
     container, model, ...(index && { index }), onRendered,
   });
 }
 
-function renderItemTemplate(model?: any, container?: any, index?: number, onRendered?: () => void): Element {
+function renderItemTemplate(
+  model?: any, container?: any, index?: number, onRendered?: () => void): Element {
```
Comment (LazyLahtak) [inline src/core/__tests__/template.test.tsx:29]:
```suggestion
  model?: any,
  container?: any,
  index?: number,
  onRendered?: () => void,
): Element {
```

File: src/core/__tests__/template.test.tsx [hunk 3]
```diff
@@ -151,8 +151,9 @@ function testTemplateOption(testedOption: string) {
     const component = mount(React.createElement(ComponentWithTemplates, elementOptions));
 
     const container = document.createElement('div');
-    renderItemTemplate({ text: 'with data'}, container);
+    renderItemTemplate({ text: 'with data' }, container);
     component.update();
+    // eslint-disable-next-line no-useless-escape
     expect(container.innerHTML).toBe('Template with data<span style=\"display: none;\"></span>');
```
Comment (LazyLahtak) [inline src/core/__tests__/template.test.tsx:166]:
```suggestion
    expect(container.innerHTML).toBe('Template with data<span style="display: none;"></span>');
```

Comment (SelyankinFyodor) [reply src/core/__tests__/template.test.tsx:166]: this change breaks test

File: tools/src/generator.ts [hunk 4]
```diff
@@ -134,7 +134,8 @@ function mapWidget(
   };
 }
 
-function extractNestedComponents(props: IComplexProp[], rawWidgetName: string, widgetName: string): INestedComponent[] {
+function extractNestedComponents(
+  props: IComplexProp[], rawWidgetName: string, widgetName: string): INestedComponent[] {
```
Comment (LazyLahtak) [inline tools/src/generator.ts:138]:
```suggestion
  props: IComplexProp[],
  rawWidgetName: string,
  widgetName: string,
): INestedComponent[] {
```

File: tools/src/generator.ts [hunk 5]
```diff
@@ -155,14 +156,16 @@ function extractNestedComponents(props: IComplexProp[], rawWidgetName: string, w
   });
 }
 
-function extractPropTypings(options: IProp[], customTypes: Record<string, ICustomType>): IPropTyping[] {
+function extractPropTypings(
+  options: IProp[], customTypes: Record<string, ICustomType>): IPropTyping[] {
```
Comment (LazyLahtak) [inline tools/src/generator.ts:160]:
```suggestion
  options: IProp[],
  customTypes: Record<string, ICustomType>,
): IPropTyping[] {
```

File: tools/src/generator.ts [hunk 6]
```diff
@@ -155,14 +156,16 @@ function extractNestedComponents(props: IComplexProp[], rawWidgetName: string, w
...
   return options
     .map((o) => createPropTyping(o, customTypes))
     .filter((t) => t != null);
 }
 
 function createPropTyping(option: IProp, customTypes: Record<string, ICustomType>): IPropTyping {
-  const isRestrictedType = (t: ITypeDescr): boolean => t.acceptableValues && t.acceptableValues.length > 0;
+  const isRestrictedType = (t: ITypeDescr): boolean => t.acceptableValues
+  && t.acceptableValues.length > 0;
```
Comment (LazyLahtak) [inline tools/src/generator.ts:168]:
```suggestion
  const isRestrictedType = (t: ITypeDescr): boolean => t.acceptableValues?.length > 0;
```

**Review by LazyLahtak — APPROVED** [review]

**Pull request merged**